### PR TITLE
Edit Post - Please run locally to make sure everything is good

### LINKED ISSUE #71: Edit dialog for existing post
We should just try to use the current Post or MakePost component and refactor it to handle the editing case. If that's too hard, then we can just make a modal for editing the contents of a post.

### PR DESCRIPTION
**Pull request by maharshmellow**
There are a lot of things in this PR and I haven't had enough sleep so please run locally and check everything 😃 

Basically, it's implementing an edit post dialog by repurposing the MakePost component. When you click the edit button, it converts the post to an edit post component where you can edit the contents and the image of the post. When you click update, it should update the list of posts as well.

### CODE REVIEW

File: src/components/Homepage.jsx [hunk 1]
```diff
@@ -11,25 +12,97 @@ import demoImage from "../images/demo-img.png";
...
+  }
+
+  handlePostCreation = (post) => {
+    // eslint-disable-next-line no-console
+    console.log(post);
+    // eslint-disable-next-line no-alert
+    alert("Post Created - Check Console");
+  }
+
+  handleEditToggle = (id) => {
```
Comment (ceciliaccwei) [inline src/components/Homepage.jsx:46]: Consider we also need to be able to edit posts in Profile page, would it better to make a new component like postList and put all the functions related to edit posts there? That way we can handle editing in multiple pages.

File: src/components/post/PostPreviewModal.jsx [hunk 2]
```diff
@@ -12,17 +12,24 @@ function PostPreviewModal(props) {
...
+    content: postContent,
+    postTime: new Date(),
+    username: "username",
+  };
+
   return (
+
     <Modal size="lg" onHide={onHide} show={show} className="post-preview-modal">
-      <Modal.Header closeButton>
+      <Modal.Header>
```
Comment (ceciliaccwei) [inline src/components/post/PostPreviewModal.jsx]: Would it be better to keep the closeButton?

**Review by ceciliaccwei — CHANGES_REQUESTED** [review]

File: src/components/Homepage.jsx (hunk 1 above)
Comment (maharshmellow) [reply src/components/Homepage.jsx:46]: Yahh I sent this in the slack message to you but I'll put that as a separate PR but it shoudn't be hard to do.

Comment (ceciliaccwei) [reply src/components/Homepage.jsx:46]: Oh right!! Sorry I totally forgot about that😖

**Review by ceciliaccwei — DISMISSED** [review]
Looks good!

File: src/components/post/PostPreviewModal.jsx (hunk 2 above)
Comment (maharshmellow) [reply src/components/post/PostPreviewModal.jsx]: done

**Review by zhaoyuxuan — APPROVED** [review]

**Pull request merged**